### PR TITLE
feat: print dex version in the logs

### LINKED ISSUE #2335: Print Dex version in the logs.
### Preflight Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/dexidp/dex/blob/master/.github/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/dexidp/dex/issues) for an issue that matches the one I want to file, without success.
- [X] I am not looking for support or already pursued the available [support channels](https://github.com/dexidp/dex/issues/new/choose) without success.

### Version

2.30.0

### Storage Type

Kubernetes

### Installation Type

Other (specify below)

### Expected Behavior

I would like to see the version of Dex in the pod logs. It is very easy and convenient way to identify the version.

### Actual Behavior

Dex version can be found out in either of 2 ways.
1. Login to the dex pod and execute `dex version`.
2. make a grpc call to the Dex API.

### Steps To Reproduce

NA

### Additional Information

NA

### Configuration

_No response_

### Logs

_No response_

### PR DESCRIPTION
**Pull request by iam-veeramalla**
Signed-off-by: iam-veeramalla <abhishek.veeramalla@gmail.com>

#### What this PR does / why we need it
Closes #2335 

#### Special notes for your reviewer
@sagikazarmark as per our discussion on https://github.com/dexidp/dex/discussions/2334

#### Does this PR introduce a user-facing change?
NA

```release-note

```

### CODE REVIEW

**Review by sagikazarmark — APPROVED** [review]
I think this is fine for now. Note that logs are not covered by the backward compatibility promise: the format may change when we decide to make logs more structured.

File: cmd/dex/serve.go [hunk 1]
```diff
@@ -91,6 +91,9 @@ func runServe(options serveOptions) error {
 	if err != nil {
 		return fmt.Errorf("invalid config: %v", err)
 	}
+
+	logger.Infof("dex version: %s", version)
```
Comment (nabokihms) [inline cmd/dex/serve.go:95]: What do you think about printing Go runtime-specific versions along with the Dex version like here?https://github.com/dexidp/dex/blob/f70015dfedfba1d2dda0031f2f8c46d3d56260fd/cmd/dex/version.go#L18-L23

Comment (iam-veeramalla) [reply cmd/dex/serve.go:95]: I dont have any problem in adding them. Please let me know if you want me to add them as part of this PR.
@sagikazarmark

Comment (nabokihms) [reply cmd/dex/serve.go:95]: In my opinion, it might be helpful.

Comment (iam-veeramalla) [reply cmd/dex/serve.go:95]: sure, I will add that :)

Comment (iam-veeramalla) [reply cmd/dex/serve.go]: @nabokihms I updated the PR. Please check if it looks good to you.

Comment (nabokihms) [reply cmd/dex/serve.go]: Great!

**Review by nabokihms — APPROVED** [review]

**Comment by iam-veeramalla** [comment]
@nabokihms do I need to do anything else to get this merged ?

**Review by sagikazarmark — APPROVED** [review]

**Pull request merged**